blog/views.py

```diff
@@ -0,0 +1,102 @@
+from django.shortcuts import render
+from blog.models import Articles ,Tags, Article_Tag_Mapping, Comments
+from django.contrib import messages
+from user.models import Users
+from django.http import HttpResponse, HttpResponseRedirect
+import random
+import string
+
+# Create your views here.
+def create(request):
+    if request.method == "POST":
+        form = (request.POST)
+        check = Users.objects.get(user_id=request.user.id)
+        slug = get_slug()
+        article = Articles(user = check, title= form["title"], body = form["body"], status = form['status'], slug = slug)
+        article.save()
+        tags = form["tags"].split(",")
+        if tags[0] != '':
+            for i in tags:
+                query_set = Tags.objects.filter(tag_name= i )
+                if query_set.exists():
+                    mapping = Article_Tag_Mapping(article_id = article.id, tag_id = query_set[0].id)
+                else:
+                    tag= Tags(tag_name = i)
+                    tag.save()
+                    mapping = Article_Tag_Mapping(article_id = article.id, tag_id = tag.id)
+                mapping.save()
+        messages.info(request,"New article has been created with title {}".format(form["title"]))
+        return HttpResponseRedirect('/home')
+    else:
+        return(render(request,
+                "createArticle.html",
+                context={"check":"check"})
+            )
+
+def view(request, title,slug):
+    article = Articles.objects.filter(title = title, slug = slug)
+    tags = Article_Tag_Mapping.objects.filter(article_id = article[0].id)
+    tag_names = []
+    for i in tags:
+        tag_names.append(Tags.objects.filter(id = i.tag_id)[0].tag_name)
+    return(render(request, "article_view.html", context = { "user_articles":article, "tags":tag_names}))
+
+def update(request, title,slug):
+    article = Articles.objects.filter(title = title, slug = slug)
```

Comment:
Instead of separately querying articles and the tag, can you think of more efficient solution